core/src/main/java/org/jdbi/v3/core/internal/OnDemandExtensions.java

```diff
@@ -37,11 +37,11 @@ public class OnDemandExtensions implements JdbiConfig<OnDemandExtensions> {
 
     static {
         try {
-            EQUALS_METHOD = Object.class.getMethod("equals", Object.class);
-            HASHCODE_METHOD = Object.class.getMethod("hashCode");
-            TOSTRING_METHOD = Object.class.getMethod("toString");
-        } catch (NoSuchMethodException wat) {
-            throw new IllegalStateException("OnDemandExtensions initialization failed", wat);
+            EQUALS_METHOD = JdbiClassUtils.methodLookup(Object.class, "equals", Object.class);
+            HASHCODE_METHOD = JdbiClassUtils.methodLookup(Object.class, "hashCode");
+            TOSTRING_METHOD = JdbiClassUtils.methodLookup(Object.class, "toString");
+        } catch (Exception e) {
+            throw new ExceptionInInitializerError(e);
```

Comment:
Is this catch necessary at all now? `methodLookup` declares no checked exceptions

Reply:
catch Exception also catches the runtime exception and so this throws ExceptionInInitializerError (which is the right thing to do in an initializer)

Reply:
That wrapping is already done by the compiler if you just leave the catch out, pretty sure.